src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java

```diff
@@ -0,0 +1,198 @@
+            arguments(CaseDataBuilderSpec.builder().atStateSpec1v2Solicitor1UnregisteredSolicitor2Registered().build())
+        );
+    }
+
+    @ParameterizedTest(name = "{index}: The state is transitioned correctly from SPEC_DRAFT to CLAIM_SUBMITTED")
+    @MethodSource("caseDataStream")
+    void shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted(CaseData caseData) {
+        // When
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
+
+        // Then Claim will go through state SPEC_DRAFT and finish at state CLAIM_SUBMITTED
+        assertThat(stateFlow.getState())
+            .extracting(State::getName)
+            .isNotNull()
+            .isEqualTo(CLAIM_SUBMITTED.fullName());
+        assertThat(stateFlow.getStateHistory())
+            .hasSize(2)
+            .extracting(State::getName)
+            .containsExactly(
+                SPEC_DRAFT.fullName(), CLAIM_SUBMITTED.fullName());
+    }
+
+    @ParameterizedTest(name = "{index}: The common state flow flags are present for a case transitioned to CLAIM_SUBMITTED")
+    @MethodSource("caseDataStream")
+    void shouldContainCommonFlags_whenCaseDataAtStateClaimSubmitted(CaseData caseData) {
+        //When
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
+
+        // Then Claim will have SPEC_RPA_CONTINUOUS_FEED, NOTICE_OF_CHANGE, GENERAL_APPLICATION_ENABLED, CERTIFICATE_OF_SERVICE and RPA_CONTINUOUS_FEED
+        assertThat(stateFlow.getFlags()).contains(
+            entry(FlowFlag.SPEC_RPA_CONTINUOUS_FEED.name(), false),
+            entry(FlowFlag.NOTICE_OF_CHANGE.name(), false),
+            entry(FlowFlag.GENERAL_APPLICATION_ENABLED.name(), false),
+            entry(FlowFlag.CERTIFICATE_OF_SERVICE.name(), false),
+            entry(FlowFlag.RPA_CONTINUOUS_FEED.name(), false)
+        );
+    }
+
+    @ParameterizedTest(name = "{index}: The state flow flag ONE_RESPONDENT_REPRESENTATIVE is set to true (for appropriate cases)")
+    @MethodSource("caseDataStreamOneRespondentRepresentative")
+    void shouldHaveOneRespondentRepresentativeFlagsSet_whenCaseDataAtStateClaimSubmitted(CaseData caseData) {
+        // When
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
+
+        // Then Claim will have ONE_RESPONDENT_REPRESENTATIVE set to true
+        assertThat(stateFlow.getFlags()).contains(
+            entry(FlowFlag.ONE_RESPONDENT_REPRESENTATIVE.name(), true)
+        );
+        assertThat(stateFlow.getFlags()).hasSize(6);    // bonus: if this fails, a flag was added/removed but tests were not updated
+    }
+
+    @ParameterizedTest(name = "{index}: The state flow flags ONE_RESPONDENT_REPRESENTATIVE and " +
+        "TWO_RESPONDENT_REPRESENTATIVES are present and set to the correct values (for appropriate cases)")
+    @MethodSource("caseDataStreamTwoRespondentRepresentatives")
+    void shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitorSpecified(CaseData caseData) {
```

Comment:
be more specific in the name like whenCaseDataAtStateClaimSubmitted1v2DifferentSol1repAnd1unreg

Reply:
no need of using specified as the test class is for specified only

Reply:
This test is for two separate 1v2 scenarios //AC 3 (1V2 different defendant solicitor) and //AC 11 1v2 def 1 represented solicitor unregistered, and def 2 solicitor registered

Reply:
Updated name as it's for testing the flag on the above scenarios (ac3 and ac11)
shouldReturnFlags_whenAtStateClaimSubmitted1v2DiffSol_OrSolicitor1UnrepAndSolicitor2Registered